microcosm_eventsource/immutable_event_factory.py

```diff
@@ -0,0 +1,81 @@
+"""
+Factory for immutable events.
+
+In extension of EventFactory, this class provides a native support of event updating container objects based
+on event types
+
+"""
+from collections import defaultdict
+from typing import Dict
+
+from microcosm_eventsource.factory import EventFactory
+
+
+container_common_mutator_registry: Dict[str, list] = defaultdict(list)
+container_event_specific_mutator_registry: Dict[str, list] = defaultdict(list)
+
+
+def register_container_mutator_common(event_type):
```

Comment:
I chime in on naming down in the tests. I had to see the usage to understand what you're doing.